packages/amplify_datastore/ios/Classes/DataStoreHubEventStreamHandler.swift

```diff
@@ -53,19 +53,23 @@ public class DataStoreHubEventStreamHandler: NSObject, FlutterStreamHandler {
         // Replay events. On hot restart, `onListen` is called again with a new listener. However,
         // DataStore will not re-emit events such as ready and modelSynced. As a result, this info
         // is lost on the Flutter side unless we replay the history prior to the hot restart.
+        #if DEBUG
         if !eventHistory.isEmpty {
             eventGuard.lock()
             defer { eventGuard.unlock() }
             for payload in eventHistory {
-                if replayableEvents.contains(payload.eventName) {
-                    sendPayload(payload)
-                }
+                sendPayload(payload)
             }
         }
+        #endif
         token = Amplify.Hub.listen(to: .dataStore) { [unowned self] (payload) in
+            #if DEBUG
```

Comment:
Should we apply similar strategy in Android implement as well?

Reply:
Yeah, we should probably do some profiling there as well